### ISSUE TITLE
Kubernetes Ingress load balancer rewrites everything to index

### ISSUE THREAD

**Issue by warent** [kubernetes/kubernetes#53846]
**TL;DR** for some reason every http service seems to have the path rewritten as `/`

I'm pretty new to Kubernetes and trying to set up an ingress load balancer.

    Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.6", GitCommit:"4bc5e7f9a6c25dc4c03d4d656f2cefd21540e28c", GitTreeState:"clean", BuildDate:"2017-09-14T06:55:55Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
    Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.6-gke.1", GitCommit:"407dbfe965f3de06b332cc22d2eb1ca07fb4d3fb", GitTreeState:"clean", BuildDate:"2017-09-27T21:21:34Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

I'm setting everything up in Google Container Engine (GKE)

It's partially working, which is a good thing. However, for some reason it's routing everything to my service as if the request is directed at `/`.

What's happening here? My guess is poor configuration.
```yaml
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
      name: main-ingress
      annotations:
        kubernetes.io/ingress.global-static-ip-name: "ingress-main-ip"
    spec:
      tls:
      - secretName: cert-main
        hosts:
        - api.example.com
      rules:
      - host: api.example.com
        http:
          paths:
          - path: /my_service/*
            backend:
              serviceName: myService
              servicePort: main-port
          - path: /my_service2/*
            backend:
              serviceName: myService2
              servicePort: main-port
```
Whenever I make a request to `api.example.com/my_service/` then myService will register the request as coming from index--no problem.

Same thing with `https://api.example.com/my_service2/` which goes to myService2

But whenever I make a request to `https://api.example.com/my_service/bogus/path/that/should/return/404` then my service seems to think the request is being directed at the `/` path again just as before.

Of course when I run myService locally (which happens to be a Go http server), then it works perfectly fine. **To be especially clear:** running this locally `127.0.0.1/bogus/path/that/should/return/404` *returns a 404 as expected* (and of course other API endpoints also work)

Here's an extremely strange thing: whenever I visit the static IP address directly  (aka ingress-main-ip, let's just say it's 1.2.3.4), for example `http://1.2.3.4/my_service/` or `http://1.2.3.4/my_service2/` it *always* returns a 404 from the GKE default backend, so my services aren't even getting routed to.

Summary:

 1. Why are the domain versions of the requests all routing to `/`
 2. Why do the direct IP requests not work at all?

Thank you for your time and help!

**Comment by warent** [comment]
This wasn't a bug, just my lack of experience :)

If anyone sees this and is wondering, please refer to here for the answer: https://stackoverflow.com/questions/46720563/kubernetes-ingress-load-balancer-rewrites-everything-to-index-why
